code/game/machinery/camera/tracking.dm

```diff
@@ -67,8 +67,36 @@
 	if(!track.initialized)
 		trackable_mobs()
 
-	var/datum/weakref/target = (isnull(track.humans[target_name]) ? track.others[target_name] : track.humans[target_name])
-	ai_start_tracking(target.resolve())
+	var/datum/weakref/target_ref = (isnull(track.humans[target_name]) ? track.others[target_name] : track.humans[target_name])
+	var/atom/target = target_ref.resolve()
+
+	attempt_track(target)
+
+/mob/living/silicon/ai/proc/attempt_track(mob/living/target)
+	//If the target has sensors on, track instantly
+	//If the AI has malf upgrades, allow instant tracking
+	var/instant_track = !!malf_picker || issilicon(target) || isbot(target)
+
+	if(!instant_track && isliving(target))
+		var/mob/living/L = target
+		if(HAS_TRAIT(L, TRAIT_SUIT_SENSORS) || HAS_TRAIT(L, TRAIT_NANITE_SENSORS))
```

Comment:
This works even if the location beacon is disabled

Reply:
This is fine actually, it only really benefits you then if you actively disable your sensors so has less impact on the AI